signer/tests/integration/postgres.rs

```diff
@@ -2310,3 +2311,501 @@ async fn can_store_and_get_latest_sweep_transaction() {
+    let test_data = TestData::generate(&mut rng, &signer_set, &test_params);
+    test_data.write_to(&db).await;
+
+    // Let's create a deposit request, we'll write it to the database
+    // later.
+    let deposit_request: model::DepositRequest = fake::Faker.fake_with_rng(&mut rng);
+    let chain_tip = db.get_bitcoin_canonical_chain_tip().await.unwrap().unwrap();
+    let txid = &deposit_request.txid;
+    let output_index = deposit_request.output_index;
+    let signer_public_key = &signer_set[0];
+
+    // The deposit request is not in our database so we should get None
+    // here.
+    let report = db
+        .get_deposit_request_report(&chain_tip, txid, output_index, signer_public_key)
+        .await
+        .unwrap();
+
+    assert!(report.is_none());
+
+    // We're going to write the deposit request to the database. We also
+    // write the deposit transaction to the database. For that transaction
+    // we want to test what happens if it is not on the canonical bitcoin
+    // transaction.
+    let random_block: model::BitcoinBlock = fake::Faker.fake_with_rng(&mut rng);
+    let tx = model::Transaction {
+        txid: deposit_request.txid.into_bytes(),
+        tx: Vec::new(),
+        tx_type: model::TransactionType::DepositRequest,
+        block_hash: random_block.block_hash.into_bytes(),
+    };
+    let tx_ref = model::BitcoinTxRef {
+        txid: deposit_request.txid,
+        block_hash: random_block.block_hash,
+    };
+
+    db.write_deposit_request(&deposit_request).await.unwrap();
+
+    // Sanity check, that if the transaction is not in our database then
+    // the report comes back empty.
+    let report = db
+        .get_deposit_request_report(&chain_tip, txid, output_index, signer_public_key)
+        .await
+        .unwrap();
+
+    assert!(report.is_none());
+
+    db.write_bitcoin_block(&random_block).await.unwrap();
+    db.write_transaction(&tx).await.unwrap();
+    db.write_bitcoin_transaction(&tx_ref).await.unwrap();
+
+    // The result shouldn't be Ok(None), since we have a deposit request,
+    // but only the amount and locktime should be present, everything else
+    // should be None.
+    let report = db
+        .get_deposit_request_report(&chain_tip, txid, output_index, signer_public_key)
+        .await
+        .unwrap()
+        .unwrap();
+
+    let report_lock_time = report.lock_time.to_consensus_u32();
+
+    assert_eq!(report.amount, deposit_request.amount);
+    assert_eq!(report_lock_time, deposit_request.lock_time);
+    assert!(report.is_accepted.is_none());
+    assert!(report.can_sign.is_none());
+    // The transaction is not on the canonical bitcoin blockchain, so it
+    // shows up as unconfirmed.
+    assert_eq!(report.status, DepositRequestStatus::Unconfirmed);
+
+    testing::storage::drop_db(db).await;
+}
+
+/// Check that if the deposit has been confirmed on a block that is not on
+/// the canonical bitcoin blockchain that the deposit reports the status as
```

Comment:
maybe nit? also in other comments below
```suggestion
/// the canonical bitcoin blockchain then the deposit reports the status as
```

Reply:
Fixed 2 in https://github.com/stacks-network/sbtc/pull/749/commits/3df3de6e2c8ec84d292b08bf03d503573a48ebac, although I'm not sure I found them all.

Reply:
last one I found: https://github.com/stacks-network/sbtc/pull/749/files#diff-0b190461b13ee668a5cea3b0b796949de1a447745d83d42bb19802de1cd2958cR2425